### ISSUE TITLE
Re-evaluate how distribution files are organized

### ISSUE THREAD

**Issue by 0xazure** [new-xkit/XKit#364]
The `Extensions/dist/` directory is starting to get a little messy, especially once #340 is merged and we start building themes from source.  We've currently got a lot of files, and some that [probably](https://github.com/new-xkit/XKit/blob/a9980bfa1d1ab04021388c1a3c687d6bbb2b8be3/Extensions/dist/extensions.js) don't belong with `<extensionId>.json` files.

This may be a little tricky because we're still supporting fallback URLs to the old XKit servers, so major changes to directory hierarchy will have to be evaluated in the context of that continued support, unless we decide enough time has passed to remove the fallback.

I am of the opinion that the 'distribution' files should be more separate from their source counterparts.  I would propose a filesystem hierarchy like the following:

```sh
.
├── Chrome/
├── Extensions/
├── Firefox/
├── Themes/
├── dev/
├── dist/
│   ├── extensions/
│   │   ├── accesskit.json
│   │   /* ... */
│   │   └── xwidgets.json
│   ├── framework_version.json
│   ├── gallery.json
│   ├── list.json
│   ├── paperboy.json
│   └── themes.json
/* ... */
└── xkit.js
```

If we _do_ end up removing the fallback, it would be a good opportunity to clean up some of the distribution files:
* `list.json` is effectively `gallery.json` but without extension description and icon information  
* `framework_version.json` is just SemVer strings that we'll probably forget to update when we release a new core version to the browser extension stores
  * it just seems to trigger a [prompt for the user to update their extension](https://github.com/new-xkit/XKit/blob/ea4617ca8931bc38951864ae62815d1acf811e6e/Extensions/xkit_preferences.js#L368-L412), but we really shouldn't be bugging the user about something the browser is supposed to do automagically
* many of these files have (to my mind) useless keys like `server`, where it usually just hardcodes `"up"` to say the server is active, instead of any real checking of server connectivity on the client side

I also think it would be a good idea to align how Extensions and Themes are organized.  That means, essentially, instead of serving a single `themes.json` file that contains the entirety of every theme's CSS, we split the themes into separate `.json` files and have a `gallery.json`-like file that lists available themes.  This would, of course, require changes to [`themes.js`](https://github.com/new-xkit/XKit/blob/ea4617ca8931bc38951864ae62815d1acf811e6e/Extensions/themes.js#L127-L168) and how it loads themes.

**Comment by Wolvan** [comment]
:+1:

**Comment by hobinjk** [comment]
All of this looks good to me. I definitely approve of splitting themes.json in the future. The useless keys are a holdover from old XKit that would need the corresponding logic removed.

**Comment by nightpool** [comment]
I agree with everything here except removing framework_version. ideally that should be updated by gulp, no?

browsers are bad and inconsistent. firefox auto-update is going to be weird in the future and I think require infrastructure on our part anyway because we're a hidden extension and I don't know if safari auto-updates at all or what the deal is there.

**Comment by 0xazure** [comment]
I've definitely changed my stance on `framework_version.json` since I've been made aware of how we're going to be handling the Firefox situation going forward.  I agree that we'll need some kind of mechanism for letting users know that there's an update they need to download.

Conceivably, we could join everything into a single file with a Gulp task and then have XKit download and cache the data in localStorage, as long a we have mechanisms both on the server (timestamping / cache busting) and client side (user-accessible force-update buttons) to force an update.
